main.py

```diff
@@ -0,0 +1,1664 @@
+    terminal_input_frame.grid(column=0, row=3, padx=10, ipadx=2, ipady=1, sticky=W+E+N+S)
+
+    terminal_input = Entry(terminal_input_frame)
+    terminal_input.config(font=(myFont, 14), background='#d9d9d9')
+    terminal_input.pack(padx=(0,5), pady=(0,8), fill=BOTH, expand=TRUE, side=LEFT)
+
+    #enter terminal button
+    terminal_enter_button = Button(terminal_input_frame, text='Enter', command=terminal_enter) 
+    terminal_enter_button.config(background=nordic_blue, activebackground='#345',
+                                 foreground='white', activeforeground='white')
+    terminal_enter_button.pack(padx=(0,5), pady=(0,8), ipadx=15, fill=None, expand=FALSE, side=LEFT)
+    button_config(terminal_enter_button)
+
+    #clear terminal button
+    terminal_clear_button = Button(terminal_input_frame, text='Clear', command=terminal_clear) 
+    terminal_clear_button.config(activebackground='#345', activeforeground='white')
+    terminal_clear_button.pack(padx=(0,5), pady=(0,8), ipadx=15, fill=None, expand=FALSE, side=LEFT)
+    button_config(terminal_clear_button)
+
+    #bind "Enter" key event with terminal_input
+    terminal_input.bind('<Return>', terminal_enter_event)  #bind enter key with event function
+
+    '''
+    For the user terminal textbox,
+    If the user types in something that is saved/valid for our 
+    program, we will print something out on the terminal.
+    (i.e. /help will output all possible commands for our interface
+          /listdevices will output all devices in that account
+          /accountinfo will output mqtt account info
+          etc.)
+    
+    If the user types in something we don't know, just do nothing.
+    Everytime the user hits "enter", print out what was typed and clear the entry box
+    '''
+
+    for widget in frame.winfo_children():
+        widget.grid(padx=5, pady=3)
+    return frame
+
+def create_left_frame(container):
+    global device_list
+    global device_info
+    global select_message
+    global device_options
+
+    frame = Frame(container)
+    frame.columnconfigure(0, weight=1)      #one column
+    for i in range(7):      
+        frame.rowconfigure(i, weight=1)    #eight rows
+    
+    frame.option_add("*Background", 'white')
+    frame.option_add("*Foreground", 'black')
+
+    '''
+    #nRF Cloud logo *****INCOMPLETE, try to make transparent
+    nrfLogo = ImageTk.PhotoImage(Image.open("nrfcloud_logo.png"))
+    logo = Label(frame, image=nrfLogo)
+    logo.config(background='#748587')
+    logo.grid(row=0, column=0, padx=(5), pady=(5), sticky=W+E+N+S)
+    '''
+
+    #(0,0) select device drop down menu at top left
+    device_options = []
+    device_list = StringVar()
+    device_list.trace('w', change_device)
+    device_list.set("Select Device...")
+   
+    device_options.append('Select Device...')
+    for device in http_get['items']:
+        if device['id'].startswith('nrf-'): #only include devices and not account device
```

Comment:
Actually, we should accept any device ID found in the account, accept for "account-<big number>" itself of course.

The reason is that only official Nordic-made devices start with "nrf-", and only when they use JITP certs.  If we use preconnect provisioning using device_credentials_installer.py, for example, the device ID could be the device-unique UUID rather than "nrf-<IMEI>", for both Nordic-made devices and customer devices (not that we plan to release this to customers).

Reply:
i think it would be better to filter on thingType, the backend puts no restrictions on naming devices.
so you could have actual devices with an `account-` prefix.

Reply:
OK, so, any device in the account whose thingType does not equal "account".  Good idea.

Reply:
![specified_subtype](https://user-images.githubusercontent.com/107961089/191140473-435d95e8-7281-4e71-b36f-21e1de08ee87.png)
Got it! Above photo was taken after changing code from "if device['id'] starts with nrf-" to "if device['subType'] != 'account'". Also certain that it works because at first I capitalized 'A' in 'account' and the dropdown included my account device. After switching it to lowercase, I get the desired results.